gamemodes/terrortown/gamemode/shared/sh_shop.lua

```diff
@@ -67,19 +181,19 @@ function shop.CanBuyEquipment(ply, equipmentId)
 		end
 	end
 
-	local team = ply:GetTeam()
+	if equipment.limited
+		and shop.IsBoughtFor(ply, equipmentId) then
+		return false, shop.statusCode.LIMITEDBOUGHT
+	end
 
 	if equipment.globalLimited
-		and BUYTABLE[equipment.id]
-		or team
-		and equipment.teamLimited
-		and TEAMS[team]
-		and not TEAMS[team].alone
-		and TEAMBUYTABLE[team]
-		and TEAMBUYTABLE[team][equipment.id]
-		or equipment.limited
-		and ply:HasBought(equipment.ClassName) then
-		return false, shop.statusCode.LIMITEDBOUGHT
+		and shop.IsGlobalBought(equipmentId) then
+		return false, shop.statusCode.GLOBALLIMITEDBOUGHT
+	end
+
+	if equipment.teamLimited
+		and shop.IsTeamBoughtFor(ply,equipmentId) then
+		return false, shop.statusCode.TEAMLIMITEDBOUGHT
```

Comment:
I think this is pretty unreadable, in our new style it should be

```suggestion
	if
		equipment.teamLimited
		and shop.IsTeamBoughtFor(ply,equipmentId)
	then
		return false, shop.statusCode.TEAMLIMITEDBOUGHT
```

or if you prefer the "current" style:
```lua
	if equipment.teamLimited
		and shop.IsTeamBoughtFor(ply,equipmentId)
	then
		return false, shop.statusCode.TEAMLIMITEDBOUGHT
```

Reply:
Did we already agree on the new style?
As far as I know we try to let a linter do all the work, so current style should be fine until we fully switch to it.

Reply:
Not sure, but your style here is pretty weird in my opinion. Maybe it is just me but that the `and` and `return` are on the same level without any sort of devider gave me the impression that the return is missing